src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15001,35 +15001,204 @@ def stack(
+                aggfunc="min",
+                fill_value=None,
+                margins=False,
+                dropna=True,
+                margins_name="All",
+                observed=False,
+                sort=sort,
+            )
+
+            # Set the correct index names based on if index name is None
+            output_index_names = qc.get_index_names()
+            output_index_names_replace_level_with_none = [
+                None
+                if isinstance(output_index_names[i], str)
+                and output_index_names[i][0:6] == "level_"  # type: ignore
+                else output_index_names[i]
+                for i in range(len(output_index_names))
+            ]
+            if None in output_index_names_replace_level_with_none:
+                qc = qc.set_index_names(output_index_names_replace_level_with_none)
+            # Set the correct columns names based on if column name is None
+            output_column_names = qc.columns.names
+            output_column_names_replace_level_with_none = [
+                None
+                if isinstance(output_column_names[i], str)
+                and output_column_names[i][0:6] == "level_"
+                else output_column_names[i]
+                for i in range(len(output_column_names))
+            ]
+            if None in output_column_names_replace_level_with_none:
+                from snowflake.snowpark.modin.pandas.dataframe import DataFrame
+
+                qc = (
+                    DataFrame(query_compiler=qc)
+                    .rename_axis(output_column_names_replace_level_with_none, axis=1)
+                    ._query_compiler
+                )
+        else:
+            # N.B. normally non-MultiIndex cases would throw an error in pandas
+            # for series objects. However, pandas seems to handle this for DFs
+            # by doing a transposed stack! Strange behavior, but since we throw
+            # an error in the modin API layer for Series without MultiIndex indices,
+            # we can assume that this must be a DataFrame. This assumption might
+            # change in the future, so this code should be changed along with it.
+            if has_index_names_none:
+                # Index name defaults to "index" after reset_index() operation if index name is None
+                index_cols = ["index"]
+            else:
+                index_cols = index_names  # type: ignore
+
+            col_label = (
+                "index_second_level" if self.columns.name is None else self.columns.name
+            )
+            qc = (
+                self.reset_index()
+                .melt(
+                    id_vars=index_cols,  # type: ignore
+                    value_vars=self.columns,
+                    var_name=col_label,
+                    value_name=MODIN_UNNAMED_SERIES_LABEL,
+                    ignore_index=False,
+                )
+                .replace(to_replace=UNPIVOT_NULL_REPLACE_VALUE, value=np.nan)
```

Comment:
here i think i still have the question about why we need to do such replace, can you give me some more details about this, if it is a bug, we should have ticket to fix it

Reply:
Repro from what I am seeing:

```
native_df = native_pd.DataFrame(data=[[0, np.nan], [np.nan, 3]], index=["cat", "dog"], columns=["weight", "height"])
snow_df = pd.DataFrame(native_df)
snow_df.reset_index().melt(id_vars="index", value_vars=['weight', 'height'], var_name="added_column", value_name="__reduced__", ignore_index=False)
```

Reply:
Filed JIRA with the above example to repro and added a comment in the code.